scripts/migration/renamings.js

```diff
@@ -1011,6 +1011,39 @@ const renamings = {
       },
     },
   },
+  '7.20211209.0': {
```

Comment:
Did we do a patch release?  If so, then this should be `.3` or whatever the last point release was, rather than `.0`, unless these changes will be included in the point release, in which case it should be whatever the last release not to include the changes was).

This question brings to light a less-than-great aspect of how I decided to version the renamings, because every time there is a point release in `master`, the corresponding entry in `develop` should be updated accordingly (unless the renamings were actually included in the release, in which case a new section should be started in the usual way).

It might have been better to label renamings with the first version they do appear in (rather than the last one they don't appear in), with the "current" in-progress release just labelled "IN PROGRESS" or something, and add a step to the quarterly release process to replace that with whatever version they are actually released with.  Thoughts?

Reply:
Just to clarify. You are thinking about the scenario where I want to release a patch with my current change. So we would have to go back and add these values to `6.x` instead of `7.x`? And if we instead did the first release that they appear, we would just have to add them to `7.x`? I think I might not be understanding correctly. It seems like in either case you have to know whether you are going to release a patch with your change and then place it in the correct location?

Reply:
I am thinking about two scenarios:

* If a patch is done which releases none of the code that contains renamings, then _at the moment_ the entry in `renamings.js` should be changed from:
  ```JS
    '7.20211209.0': {
  ```
  to
  ```JS
    '7.20211209.1': {
  ```
  (or in this case `.2` before and `.3` after, since I _think_ we've already done a couple of patch releases), because _these renamings will take effect after the patch release_.

* If a patch is done which releases some/all of the renamings, then this file should instead be changed to:

That is how things work with the versioning scheme I originally proposed and which we have been using.  But maybe it would be better to change it, such that instead it would be versioned like this:

  ```JS
    '7.20211209.1': {
      // Renamings that are released in 7.20211209.1.
    },
    'DEVELOP': {
      // Renamings that will not be released until after 7.20211209.1.
    },
  ```

Where `DEVELOP` would be replaced by the actual version number during the release process.

Reply:
(In any case, this discussion is tangential to this PR, and should not cause it to be delayed.)

Reply:
Sorry I am just getting back to this. Ah that makes sense, thanks for clarifying. I think having 'develop' is a good idea and it should be pretty easy to automate swapping out 'develop' with the release number when we run  `npm run recompile`.